synapse/storage/databases/main/event_federation.py

```diff
@@ -670,24 +671,61 @@ def get_oldest_events_with_depth_in_room_txn(self, txn, room_id):
 
         return dict(txn)
 
-    async def get_max_depth_of(self, event_ids: List[str]) -> int:
-        """Returns the max depth of a set of event IDs
-
+    async def get_max_depth_of(self, event_ids: List[str]) -> Tuple[str, int]:
```

Comment:
A bit of new logic here. We also return the event ID for the event with the greatest depth

Reply:
Presumably this new logic will get used when we add the bulk insert API? (It's not used in this PR anyomre)

Reply:
It is still used in https://github.com/matrix-org/synapse/pull/9247. I can remove here if preferred.

Reply:
I think its ok to leave here.